subgraph/queries/tokens.py

```diff
@@ -101,86 +95,86 @@ def fetch_token_balances(
+            console.log(
+                f"Fetching {len(next_page['tokenBalances'])} token balances"
+            )
+            for entry in next_page["tokenBalances"]:
+                address = entry["id"].split("-")[0]
+                amount = int(entry["balance"])
+                if amount > 0:
+                    if entry["token"]["symbol"] == "BADGER":
+                        badger_balances[address] = amount / DECIMAL_MAPPING[chain]
+                    if entry["token"]["symbol"] == "DIGG":
+                        fragment_balance = digg_utils.shares_to_fragments(
+                            int(amount)
+                        )
+                        digg_balances[address] = float(fragment_balance) / 1e9
     return badger_balances, digg_balances
 
 
-# flake8: noqa: E501
-def fetch_fuse_pool_balances(client: Client, chain: Network, block: int) -> Dict[str, Dict]:
-    if chain != Network.Ethereum:
-        console.log("Fuse pools are only active on ETH")
-        return {}
-
-    client = SubgraphClient("fuse", chain)
-    ctoken_data = {
-        "fBADGER-22": {
-            "underlying_contract": "0x3472A5A71965499acd81997a54BBA8D852C6E53d",
+def fetch_fuse_token_info() -> Dict:
+    return {
+        BADGER: {
+            "underlying": BADGER,
             "symbol": "fBADGER-22",
-            "contract": "0x6780B4681aa8efE530d075897B3a4ff6cA5ed807",
+            "contract": FBADGER,
         },
-        "fDIGG-22": {
-            "underlying_contract": "0x798D1bE841a82a273720CE31c822C61a67a601C3",
+        DIGG: {
+            "underlying": DIGG,
             "symbol": "fDIGG-22",
-            "contract": "0x792a676dD661E2c182435aaEfC806F1d4abdC486",
+            "contract": FDIGG
         },
+        BVECVX: {
+            "underlying": BVECVX,
+            "symbol": "fBVECVX-22",
+            "contract": FBVECVX
+        }
     }
 
-    for symbol, data in ctoken_data.items():
-        ftoken = make_contract(
-            Web3.toChecksumAddress(data["contract"]),
-            abi_name=Abi.CErc20Delegator,
-            chain=chain,
-        )
-        underlying = make_contract(
-            Web3.toChecksumAddress(data["underlying_contract"]),
-            abi_name=Abi.ERC20,
-            chain=chain,
-        )
-
-        ctoken_data[symbol]["decimals"] = int(ftoken.decimals().call())
-        console.log(ctoken_data[symbol]["decimals"])
-        underlying_decimals = int(underlying.decimals().call())
-        mantissa = 18 + underlying_decimals - ctoken_data[symbol]["decimals"]
-        ctoken_data[symbol]["exchange_rate"] = float(
-            ftoken.exchangeRateStored().call()
-        ) / math.pow(10, mantissa)
 
+def fetch_fuse_pool_token(chain: Network, block: int, token: str) -> Dict[str, Decimal]:
+
+    if chain != Network.Ethereum:
+        return {}
+
+    fuse_client = SubgraphClient("fuse", chain)
+    fuse_token_info = fetch_fuse_token_info()
+    if token not in fuse_token_info:
```

Comment:
How can `fuse_token_info` be None if func always returns a dict?

Reply:
This is to check just in case the function gets called with a token that is not available, will change fuse tokens to constant